### PR TITLE
Feature/core.loader data access layer configure

### PR DESCRIPTION
**Pull request by VladiTitov**
added and configured DataAccess layer in project, created factory for database providers, added database context on Dapper framework, created migration logic

### CODE REVIEW

**Review by Egorkas — APPROVED** [review]
good work!

File: Octopus.Core.Common/DynamicObject/Services/DynamicObjectCreateService.cs [hunk 1]
```diff
@@ -11,14 +11,18 @@ namespace Octopus.Core.Common.DynamicObject.Services
...
         private IList<DynamicProperty> _dynamicProperties;
+        private readonly IJsonDeserializer _jsonDeserializer;
         private readonly IDynamicTypeFactory _dynamicTypeFactory;
 
         public DynamicObjectCreateService(IJsonDeserializer jsonDeserializer, IDynamicTypeFactory dynamicTypeFactory)
         {
             _dynamicTypeFactory = dynamicTypeFactory;
-            _dynamicProperties = jsonDeserializer.GetDynamicProperties<DynamicProperty>(@"Configs\dynamicProperties.json");
+            _jsonDeserializer = jsonDeserializer;
+            _dynamicProperties = GetDynamicProperties(@"Configs\dynamicProperties.json");
```
Comment (kogonidze) [inline Octopus.Core.Common/DynamicObject/Services/DynamicObjectCreateService.cs:21]: Don`t you want to pass the path to file with dynamicProperties description as a parameter to function CreateTypeByDescription? As it implemented in my version of dynamic object service

Comment (VladiTitov) [reply Octopus.Core.Common/DynamicObject/Services/DynamicObjectCreateService.cs]: fixed

**Review by kogonidze — APPROVED** [review]
Looks good

**Pull request merged**